elements/pf-listbox/pf-listbox.ts

```diff
@@ -0,0 +1,362 @@
+
+  /**
+   * all slotted listbox options and/or groups of options
+   */
+  @queryAssignedElements() private groupsOrOptions!: PfListboxGroupOrOption[];
+
+  #internals = new InternalsController(this, {
+    role: 'listbox'
+  });
+
+  #tabindex = new RovingTabindexController<PfListboxOption>(this);
+
+  /**
+   * all options that will not be hidden by a filter
+   * */
+  #_allOptions: PfListboxOption[] = [];
+
+  #shiftStartingItem: PfListboxOption | null = null;
+
+  get isHorizontal(): boolean {
+    return this.#internals.ariaOrientation === 'horizontal';
+  }
+
+  get isMultiselectable(): boolean {
+    return this.#internals.ariaMultiSelectable === 'true';
+  }
+
+  get activeItem() {
+    const [active] = this.#allOptions.filter(option => option.getAttribute('id') === this.#internals.ariaActivedescendant);
+    return active || this.#tabindex.firstItem;
+  }
+
+  get #allOptions() {
+    return this.#_allOptions;
+  }
+
+  set #allOptions(options: PfListboxOption[]) {
+    this.#_allOptions = this.filterMode === 'required' && this.filter === '' ? [] : options.filter(option => (this.constructor as typeof PfListbox).isOption(option) && (this.filterMode === 'disabled' || option.getUpdateByFilter(this.filter)));
+  }
+
+  set value(items: string | null) {
+    const oldValue = this.value;
+    const selectedItems = items?.toLowerCase().split(',');
+    const [firstItem] = selectedItems || [null];
+    this.options.forEach(option => {
+      const textContent = (option.textContent || '').toLowerCase();
+      const selected = this.isMultiselectable ? selectedItems?.includes(textContent) : firstItem === textContent;
+      option.ariaSelected = `${selected}`;
+    });
+    if (oldValue !== this.value) {
+      this.#fireInput();
+    }
+  }
+
+  get value() {
+    const selectedItems = this.options.filter(option=>option.ariaSelected === 'true').map(option => option.textContent);
+    return selectedItems.join(',');
+  }
+
+  get validOptions() {
+    return this.options;
+  }
+
+  isValid(val: string | null) {
+    const vals = val?.split(',') || [];
+    const options = this.options.map(option => option.textContent);
+    return vals.every(val => {
+      return options.includes(val);
+    });
+  }
+
+  updated(changed: PropertyValues<this>) {
+    if (changed.has('filter')) {
+      this.#onFilterChange();
+    }
+    if (changed.has('orientation')) {
+      this.#internals.ariaOrientation = ['vertical', 'horizontal'].includes(this.orientation) ? this.orientation : null;
+    }
+    if (changed.has('multiple')) {
+      this.#internals.ariaMultiSelectable = this.multiple ? 'true' : 'false';
+    }
+  }
+
+  render() {
+    return html`
+      <slot 
+        class="${this.isHorizontal ? 'horizontal' : 'vertical'}"
+        @slotchange="${this.#onSlotchange}" 
+        @optionfocus="${this.#onOptionFocus}"
+        @click="${this.#onOptionClick}"
+        @keydown="${this.#onOptionKeydown}"
+        @keyup="${this.#onOptionKeyup}">
```

Comment:
in the absence of `<pf-listbox>`, we could share code here by snipping an [element-position directive](https://lit.dev/docs/templates/custom-directives/#parts) on some hypothetical ListboxController

```ts
html`
  <slot class="..." ${this.#listbox.listbox()}></slot>

Reply:
Moved function to `listbox-controller`. Since we so have a `simple-list` (and several other components that use it) upstream, I renamed `pf-listbox` to `simple-list` in case we want to keep it.